### DISCUSSION TITLE
Includes: is it possible to use it, to include file text in block code?

### DISCUSSION THREAD

**Discussion by aborruso** [quarto-dev/quarto-cli#874] [Q&A] [answered]
Hi,
I would like to include code in a revealjs presentation, starting from an external file.

Is there a way to do it? Is there a way to run something like the code below?

Thank you

~~~
```yaml
{{< include _content.yml>}}
```
~~~

**Comment by jjallaire** [comment]
For this you use the `metadata-file` option:

````yaml
---
metadata-file: _content.yml
---
````

**Reply by aborruso** [reply]
@jjallaire thank you, but I'm not sure I understand you.

I need to insert in a slide some code (not calculated), something like the image below.

![image](https://user-images.githubusercontent.com/30607/168423683-e3473e79-435f-4abc-9b99-3f445c8025c2.png)


Normally I write

~~~
```{.json}
{"hello":"world"}
```
~~~


I would like to not hard code it, but simply including it from an external file. Do I must write this?

~~~
```{.json}
---
metadata-file: file.json
---
```
~~~

**Comment by jjallaire** [comment, marked as answer]
Okay, we don't have a feature that does this elegantly right now but you can use the include-code-files filter to get this behavior: https://github.com/pandoc/lua-filters/tree/master/include-code-files

**Reply by aborruso** [reply]
It works great, thank you.
